src/main/java/flaxbeard/steamcraft/block/BlockBoiler.java

```diff
@@ -237,13 +237,14 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer p
     	TileEntityBoiler tileEntity = (TileEntityBoiler) world.getTileEntity(x, y, z);
 
         boolean isClient = !world.isRemote;
-        
-	    if (!FluidHelper.playerIsHoldingWaterContainer(player) && isClient && tileEntity != null) {
-    		player.openGui(Steamcraft.instance, 0, world, x, y, z);
-        } else {
-        	FluidHelper.fillTankFromHeldItem(player, tileEntity.myTank);
+        if (tileEntity != null) {
+        	if (!FluidHelper.playerIsHoldingWaterContainer(player) && isClient && tileEntity != null) {
```

Comment:
I now notice that 'tileEntity'  null checking  is superfluous  as it already exists one line below.